File: dagger2demo/src/main/java/com/example/dagger2demo/module/HttpModule.java
package com.example.dagger2demo.module;

import com.example.dagger2demo.object.HttpObject;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

/**
 * Created by zhengwei on 2021/5/23. 这个类用来提供对象
 */
@Singleton
@Module
public class HttpModule {

    @Singleton
    @Provides
    public HttpObject providerHttpObject() {
        return new HttpObject();
    }

}

File: hiltdemo/src/main/java/com/example/hiltdemo/HttpModule.kt
package com.example.hiltdemo

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ApplicationComponent

/**
 * Created by zhengwei on 2021/5/23.
 */
@InstallIn(ApplicationComponent::class)
@Module
class HttpModule {
    @Provides
    fun getHttpObject(): HttpObject {
        return HttpObject()
    }
}